model.py

```diff
@@ -2671,13 +2561,12 @@ def apply_similarity_threshold(self, candidates, threshold=0.5):
 
     def best_cover_within_distance(self, distance, threshold=0.5):
         _db = Session.object_session(self)
-        flattened_data = [self.primary_identifier.id]
+        identifier_ids = [self.primary_identifier.id]
         if distance > 0:
-            data = Identifier.recursively_equivalent_identifier_ids(
-                _db, flattened_data, distance, threshold=threshold)
-            flattened_data = Identifier.flatten_identifier_ids(data)
+            identifier_ids = Identifier.recursively_equivalent_identifier_ids(
+                _db, identifier_ids, distance, threshold=threshold)
 
-        return Identifier.best_cover_for(_db, flattened_data)
+        return Identifier.best_cover_for(_db, identifier_ids)
```

Comment:
There are a number of places like this where we get a list of identifier ids and pass it to something else. It would be faster to use a subquery, but that can be a future branch.